src/jit/instrsxarch.h

```diff
@@ -180,9 +180,10 @@ INSTMUL(imul_15, "imul", 0, IUM_RD, 0, 1, BAD_CODE, 0x4400003868, BAD_CODE)
 INST3(FIRST_SSE2_INSTRUCTION, "FIRST_SSE2_INSTRUCTION",  0, IUM_WR, 0, 0, BAD_CODE, BAD_CODE, BAD_CODE)
 
 // These are the SSE instructions used on x86
-INST3( mov_i2xmm,   "movd"        , 0, IUM_WR, 0, 0, BAD_CODE,     BAD_CODE, PCKDBL(0x6E)) // Move int reg to a xmm reg. reg1=xmm reg, reg2=int reg 
-INST3( mov_xmm2i,   "movd"        , 0, IUM_WR, 0, 0, BAD_CODE,     BAD_CODE, PCKDBL(0x7E)) // Move xmm reg to an int reg. reg1=xmm reg, reg2=int reg 
+INST3( mov_i2xmm,   "movd"        , 0, IUM_WR, 0, 0, BAD_CODE,     BAD_CODE, PCKDBL(0x6E)) // Move int reg to a xmm reg. reg1=xmm reg, reg2=int reg
+INST3( mov_xmm2i,   "movd"        , 0, IUM_WR, 0, 0, BAD_CODE,     BAD_CODE, PCKDBL(0x7E)) // Move xmm reg to an int reg. reg1=xmm reg, reg2=int reg
 INST3( pmovmskb,    "pmovmskb"    , 0, IUM_WR, 0, 0, BAD_CODE,     BAD_CODE, PCKDBL(0xD7)) // Move the MSB bits of all bytes in a xmm reg to an int reg
+INST3( movmskpd,    "movmskpd"    , 0, IUM_WR, 0, 0, BAD_CODE,     BAD_CODE, PCKDBL(0x50)) // Extract 2-bit sign mask from xmm and store in reg. The upper bits of r32 or r64 are filled with zeros.
```

Comment:
You have the entry in the RM field. This particular instruction is only ever RR

Reply:
I have decided to follow prior art - definitions of identical instructions handling bytes;

![pmovmskb](https://user-images.githubusercontent.com/801959/35392950-d290d9c6-01e3-11e8-98f7-e258be28191b.PNG)

![movmskpd](https://user-images.githubusercontent.com/801959/35392963-decdf9b2-01e3-11e8-9ad8-41d90f14c464.PNG)

Reply:
@CarolEidt?

Reply:
I'm not great at recalling the details of the instruction encodings, but my recollection is that if it is spec'd as being RM, then we need to use the RM field even if it only takes registers, so that the encoding is done as RM. Perhaps @fiigii can comment?

Reply:
![image](https://user-images.githubusercontent.com/10487869/35397490-79561a7c-01a4-11e8-9e51-f0e39a151bbd.png)

The table for `movmskpd` is:
![image](https://user-images.githubusercontent.com/10487869/35397453-62095e7e-01a4-11e8-8ba4-7f964da5077f.png)

Reply:
Looks like we aren't actually using the `rr` entry for anything other than `inc`/`dec` and `push`/`pop`, however.

```
/*****************************************************************************
 *  x86 instructions for  the JIT compiler
 *
 *          id      -- the enum name for the instruction
 *          nm      -- textual name (for assembly dipslay)
 *          fp      -- 1 = floating point instruction, 0 = not floating point instruction
 *          um      -- update mode, see IUM_xx enum (rd, wr, or rw)
 *          rf      -- 1 = reads flags, 0 = doesn't read flags
 *          wf      -- 1 = writes flags, 0 = doesn't write flags
 *          mr      -- base encoding for R/M[reg] addressing mode
 *          mi      -- base encoding for R/M,icon addressing mode
 *          rm      -- base encoding for reg,R/M  addressing mode
 *          a4      -- base encoding for eax,i32  addressing mode
 *          rr      -- base encoding for register addressing mode
 *
```

Reply:
> but my recollection is that if it is spec'd as being RM, then we need to use the RM field even if it only takes registers, 

Right.